src/parser_urdf.cc

```diff
@@ -2663,48 +2665,137 @@ void URDF2SDF::ListSDFExtensions(const std::string &_reference)
+                    << "ParserConfig::URDFPreserveFixedJoint is false, could "
+                    << "help resolve this warning. See [URL_HERE] for more "
+                    << "information about this behavior.";
+        nonJointReductionErrors.emplace_back(
+            ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+        errorStream.str(std::string());
+      }
+
+      errorStream << "urdf2sdf: parent joint["
+                  << _link->parent_joint->name
+                  << "] ignored.";
+      nonJointReductionErrors.emplace_back(
+          ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+      errorStream.str(std::string());
     }
 
-    if (_link->parent_joint)
+    // check child joints for any possibility of joint lumping and reduction
+    if (!jointReductionHappens && !_link->child_joints.empty())
     {
-      sdfdbg << "urdf2sdf: link[" << _link->name
-             << "] has no inertia, "
-             << "parent joint [" << _link->parent_joint->name
-             << "] ignored.\n";
+      for (const auto &cj : _link->child_joints)
+      {
+        if (FixedJointShouldBeReduced(cj) && g_reduceFixedJoints)
+        {
+          jointReductionHappens = true;
+          break;
+        }
+        else if (cj->type == urdf::Joint::FIXED &&
+            (!FixedJointShouldBeReduced(cj) || !g_reduceFixedJoints))
+        {
+          errorStream << "urdf2sdf: allowing joint lumping by removing any "
+                      << "<disableFixedJointLumping> or <preserveFixedJoint> "
+                      << "gazebo tag on fixed child joint["
+                      << cj->name
+                      << "], as well as ensuring that "
+                      << "ParserConfig::URDFPreserveFixedJoint is false, could "
+                      << "help resolve this warning. See [URL_HERE] for more "
+                      << "information about this behavior.";
+          nonJointReductionErrors.emplace_back(
+              ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+          errorStream.str(std::string());
+        }
+      }
+
+      // none of the child joints can be lumped and reduced
+      if (!jointReductionHappens)
+      {
+        errorStream << "urdf2sdf: ["
+                    << static_cast<int>(_link->child_joints.size())
+                    << "] child joints ignored.";
+        nonJointReductionErrors.emplace_back(
+            ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+        errorStream.str(std::string());
+        errorStream << "urdf2sdf: ["
+                    << static_cast<int>(_link->child_links.size())
+                    << "] child links ignored.";
+        nonJointReductionErrors.emplace_back(
+            ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+        errorStream.str(std::string());
+      }
     }
 
-    sdfdbg << "urdf2sdf: link[" << _link->name
-           << "] has no inertia, not modeled in sdf\n";
-    return;
```

Comment:
FYI, the bug fix in this PR is due to how the logic around this `return` has changed. Previously, we simply returned if a link is massless preventing the code the recurses (line 2710) into the children of the link from running. In this PR, we only return if the massless link is not able to be lumped into its parent or child link.

Reply:
yup! That's why I was using a `sdf::Errors` container to collect all the warnings, to decide whether to print it all out at the end